### ISSUE TITLE
Upload progress event

### ISSUE THREAD

**Issue by Aizenko** [balderdashy/skipper#43]
With  
> req.file('file').on('progress', function(event) { sails.log.debug(event) }).upload(function ....

I got trace containing
written: 3230,  total: 3230,  percent: 100,
written: 5832390,  total: 5832390,  percent: 100,

percent is always 100% and It doesn't seem to have the file size (total).

Regards,

**Comment by sailsbot** [comment]
Thanks for posting, @Aizenko.  I'm a repo bot-- nice to meet you!

It has been 60 days since there have been any updates or new comments on this page.  If this issue has been resolved, feel free to disregard the rest of this message.  On the other hand, if you are still waiting on a patch, please:

  + review our [contribution guide](https://github.com/balderdashy/sails/blob/master/CONTRIBUTING.md) to make sure this submission meets our criteria (only _verified bugs_ with documented features, please;  no questions, commentary, or bug reports about undocumented features or unofficial plugins)
  + create a new issue with the latest information, including updated version details with error messages, failing tests, and a link back to [the original issue]().  This allows GitHub to automatically create a back-reference for future visitors arriving from search engines.

Thanks so much for your help!

**Comment by cayasso** [comment]
Hi guys, any update on this issue, this is making the uploader progress functionality useless  :(

**Comment by crobinson42** [comment]
Hey everyone, I have a temporary work-around for this issue. The body-parser in sails smartly separates any params passed in the request that exist BEFORE the file(s). I use `FileReader` on the front-end to get the size of the file, then I can safely calculate the upload progress based on `written / fileSize`. Hope that helps anyone that's stuck.
